src/transformStateWithClones.js

```
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state }; // копія початкового стану
+  const history = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        currentState = { ...currentState };
+
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+        break;
+
+      default:
+        break;
```

The `default` case in the `switch` statement does not handle errors for unknown action types. This violates checklist item: "The switch statement must always include a default case for error handling." You should throw an error or handle it appropriately in the default case.